--- a/cmake/local.cmake	2026-07-06 12:13:41.141873612 +0200
+++ b/cmake/local.cmake	2026-07-06 12:13:51.589768325 +0200
@@ -483,7 +483,7 @@
     get_filename_component(OLLAMA_GO_OUTPUT_DIR "${OLLAMA_GO_OUTPUT}" DIRECTORY)
 
     set(OLLAMA_GO_LDFLAGS
-        "-s -w -X=github.com/ollama/ollama/version.Version=${OLLAMA_VERSION} -X=github.com/ollama/ollama/server.mode=release")
+        "-X=github.com/ollama/ollama/version.Version=${OLLAMA_VERSION} -X=github.com/ollama/ollama/server.mode=release")
     if(GO_EXECUTABLE)
         add_custom_target(ollama-go ALL
             COMMAND ${CMAKE_COMMAND} -E make_directory "${OLLAMA_GO_OUTPUT_DIR}"
