--- a/auto/depends.sh	2026-05-28 09:53:44.956682763 +0200
+++ b/auto/depends.sh	2026-05-28 09:54:48.638104360 +0200
@@ -61,30 +61,6 @@
     fi
     exit $ret;
 fi
-gcc --version >/dev/null 2>/dev/null; ret=$?; if [[ 0 -ne $ret ]]; then
-    if [[ $OS_IS_CENTOS == YES ]]; then
-        echo "Please install gcc by:"
-        echo "  yum install -y gcc"
-    elif [[ $OS_IS_UBUNTU == YES ]]; then
-        echo "Please install gcc by:"
-        echo "  apt install -y gcc"
-    else
-        echo "Please install gcc"
-    fi
-    exit $ret;
-fi
-g++ --version >/dev/null 2>/dev/null; ret=$?; if [[ 0 -ne $ret ]]; then
-    if [[ $OS_IS_CENTOS == YES ]]; then
-        echo "Please install g++ by:"
-        echo "  yum install -y gcc-c++"
-    elif [[ $OS_IS_UBUNTU == YES ]]; then
-        echo "Please install g++ by:"
-        echo "  apt install -y g++"
-    else
-        echo "Please install gcc-c++"
-    fi
-    exit $ret;
-fi
 make --version >/dev/null 2>/dev/null; ret=$?; if [[ 0 -ne $ret ]]; then
     if [[ $OS_IS_CENTOS == YES ]]; then
         echo "Please install make by:"
@@ -143,18 +119,6 @@
 fi
 # Check tclsh, which is depended by SRT.
 if [[ $SRS_SRT == YES ]]; then
-    tclsh <<< "exit" >/dev/null 2>&1; ret=$?; if [[ 0 -ne $ret ]]; then
-        if [[ $OS_IS_CENTOS == YES ]]; then
-            echo "Please install tclsh by:"
-            echo "  yum install -y tcl"
-        elif [[ $OS_IS_UBUNTU == YES ]]; then
-            echo "Please install tclsh by:"
-            echo "  apt install -y tclsh"
-        else
-            echo "Please install tclsh"
-        fi
-        exit $ret;
-    fi
     cmake --version >/dev/null 2>/dev/null; ret=$?; if [[ 0 -ne $ret ]]; then
         if [[ $OS_IS_CENTOS == YES ]]; then
             echo "Please install cmake by:"
