--- a/3rdparty/srt-1-fit/CMakeLists.txt	2026-05-21 13:43:37.873026311 +0200
+++ b/3rdparty/srt-1-fit/CMakeLists.txt	2026-05-21 13:44:00.362762482 +0200
@@ -7,7 +7,7 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 #
 
-cmake_minimum_required (VERSION 2.8.12 FATAL_ERROR)
+cmake_minimum_required (VERSION 3.5 FATAL_ERROR)
 set (SRT_VERSION 1.5.3)
 
 set (CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/scripts")
