--- a/3rdparty/srt-1-fit/CMakeLists.txt	2026-05-16 18:19:57.838859791 +0200
+++ b/3rdparty/srt-1-fit/CMakeLists.txt	2026-05-16 18:20:22.922576984 +0200
@@ -7,7 +7,7 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 #
 
-cmake_minimum_required (VERSION 2.8.12 FATAL_ERROR)
+cmake_minimum_required (VERSION 3.5 FATAL_ERROR)
 set (SRT_VERSION 1.5.1)
 
 set (CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/scripts")
