--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,7 @@
 #
 #============================ end_copyright_notice =============================
 
-cmake_minimum_required (VERSION 3.5)
+cmake_minimum_required (VERSION 3.14)
 
 #################################################################################
 # PROJECT
@@ -622,7 +622,6 @@
         -Werror=format-security       # treats format string security warnings as errors
         -Wno-sequence-point           # disable undefined behavior warning
         -Wno-comment                  # allow multi-line comments
-        -Ofast                        # optimize code in the favor of speed 
         -fstack-protector-strong      # enables stack protection for vulnerable functions that contain: An array of any size and type
         -Wno-unused-but-set-parameter # disable parameter set but not used warning
         )
