# needed for external pager, bug #57410
--- a/sys/unix/Makefile.top
+++ b/sys/unix/Makefile.top
@@ -100,7 +100,7 @@ SPEC_LEVS = asmodeus.lua baalz.lua bigrm-*.lua castle.lua fakewiz?.lua \
 	astral.lua air.lua earth.lua fire.lua water.lua hellfill.lua tut-?.lua
 QUEST_LEVS = ???-goal.lua ???-fil?.lua ???-loca.lua ???-strt.lua
 
-DATNODLB = $(VARDATND) license symbols
+DATNODLB = $(VARDATND) license symbols $(DATHELP)
 DATDLB = $(DATHELP) dungeon.lua tribute $(SPEC_LEVS) $(QUEST_LEVS) $(VARDATD)
 DAT = $(DATNODLB) $(DATDLB)
 
