From 1a3ad8669f6c8f4288941487b4f2dba125c9f420 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Sun, 17 May 2026 19:28:49 +0200
Subject: [PATCH] Raise minimum CMake version to 3.10

Signed-off-by: Andreas Sturmlechner <asturm@gentoo.org>
---
 SuiteSparse_metis/GKlib/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SuiteSparse_metis/GKlib/CMakeLists.txt b/SuiteSparse_metis/GKlib/CMakeLists.txt
index 9cd1b4b..f15507f 100644
--- a/SuiteSparse_metis/GKlib/CMakeLists.txt
+++ b/SuiteSparse_metis/GKlib/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.10)
 project(GKlib)
 
 get_filename_component(abs "." ABSOLUTE)
-- 
2.54.0

